src/Symfony/Bundle/DebugBundle/DependencyInjection/Compiler/HtmlDumperPass.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\DebugBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Registers the file link format for the {@link \Symfony\Component\VarDumper\Dumper\HtmlDumper}.
+ *
+ * @author Christian Gripp <mail@core23.de>
+ */
+final class HtmlDumperPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('var_dumper.html_dumper')) {
+            return;
+        }
+
+        $dumperDefinition = $container->getDefinition('var_dumper.html_dumper');
+
+        if ($container->hasDefinition('debug.file_link_formatter')) {
+            $formatterDefintion = $container->getDefinition('debug.file_link_formatter');
+
+            $dumperDefinition->addMethodCall('setDisplayOptions', array(
+                array(
+                    'fileLinkFormat' => $formatterDefintion,
```

Comment:
this is wrong. You must use a reference, not an inline definition (which would create another instance of the class, as it would be a different service)